docs ($compile): add error documentation for noslot error in $compile

JoshuaJWilborn opened the pull request:
there was no error page for the $compile:noslot error

this resolves #15790

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**



**What is the current behavior? (You can also link to an open issue here)**



**What is the new behavior (if this is a feature change)?**



**Does this PR introduce a breaking change?**



**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**Other information**:

Commit:
docs ($compile): add error documentation for noslot error in $compile

there was no error page for the $compile:noslot error

this resolves #15790

JoshuaJWilborn commented:
It says that I need CLA but I've already filled one out.

JoshuaJWilborn commented:
I checked and it was because my git signature was for my work email. I've added my work email as an alternate email to my google account, how do I get the CLA bot to rerun the check? Thanks.

JoshuaJWilborn commented:
Is there anything else I need to do/update etc to get this merged? :)

Narretz commented:
I will review the PR this week

Narretz commented on docs/content/error/$compile/noslot.ngdoc:
````diff
@@ -0,0 +1,45 @@
+```
+
+```js
+\\ if we the following change we will no longer get no slot error.
````
If we make ...

Narretz reviewed (changes requested):
Looking good generally, just a few things to improve / move around

JoshuaJWilborn committed:
docs ($compile): update error doc text

removed styling
moved comments outside of code blocks
removed unneeded directive in example

JoshuaJWilborn committed:
docs ($compile): fix grammar

insert missing "make" to fix grammar

JoshuaJWilborn committed:
docs ($compile): fix name to match error

"no slot" -> "noslot"

JoshuaJWilborn committed:
docs ($compile): remove style tag from example

remove second style tag from example

Narretz committed:
Fix small typos,  and line length, and add link

Narretz commented:
Thanks @JoshuaJWilborn - for future PRs, additional commits with adjustments don't have to have a "perfect commit message" because we will squash them into one in the end. It's also a good idea to create a new branch for each PR.
